lahja/endpoint.py

```diff
@@ -202,9 +227,33 @@ def start_serving_nowait(self,
         await asyncio.gather(
             self._receiving_loop_running.wait(),
             self._internal_loop_running.wait(),
+            self._server_running.wait(),
             loop=self.event_loop
         )
 
+    async def _start_server(self) -> None:
+        self._server = await asyncio.start_unix_server(
+            self._accept_conn,
+            path=str(self.ipc_path),
+        )
+        self._server_running.set()
+
+    async def _accept_conn(self, reader: StreamReader, writer: StreamWriter) -> None:
+        remote = RemoteEndpoint(reader, writer)
+        coro = asyncio.ensure_future(self._handle_client(remote))
+        self._child_coros.add(coro)
+
+    async def _handle_client(self, remote: RemoteEndpoint) -> None:
+        while self._running and remote.more_to_read:
+            message = await remote.read_message()
+
+            if isinstance(message, Broadcast):
+                await self._receiving_queue.put((message.event, message.config))
+            else:
+                self.logger.warning(
+                    f'[{self.ipc_path}] received unexpected message: {message}'
+                )
```

Comment:
It would be nice if this could cleanup it's own `coro` from `self._child_coros` so that set doesn't end up with a bunch of finished/done/cancelled coros hanging around.  Not entirely clear how we'd go about doing that but ideas off top of my head are:

- use a mapping of `RemoteEndpoint -> coro` so this function can know which one to reference.
- use [`Future.add_done_callback`](https://docs.python.org/3/library/asyncio-future.html#asyncio.Future.add_done_callback) to manage the cleanup...
- maybe use a `weakref.WeakKeyDictionary` so that we can just track the `RemoteEndpoint` instance and once we ditch it the future cleans up automatically.